components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -1075,20 +1075,24 @@ def wrapObjectTemplate(templateBody, nullValue, isDefinitelyObject, type,
     if type.isObject():
         assert not isEnforceRange and not isClamp
 
-        # TODO: Need to root somehow
-        # https://github.com/servo/servo/issues/6382
+        templateBody = "${val}.get().to_object()"
         default = "ptr::null_mut()"
-        templateBody = wrapObjectTemplate("${val}.get().to_object()",
-                                          default,
-                                          isDefinitelyObject, type, failureCode)
 
-        if isMember in ("Dictionary", "Union"):
+        # TODO: Do we need to do the same for dictionaries?
```

Comment:
Yes.

Reply:
So I just checked and currently dictionaries are flagged as `#[must_root]` and their constructor first allocates a dict using `RootedTraceableBox` with defaults and only then sets values.
With this approach having `Heap<*mut JSObject>` as dictionary members suffices and moving the dict box is safe, as long as it's not consumed to retrieve the inner value.

Should I still pursue this, then?

Reply:
See https://github.com/servo/servo/pull/20265/commits/3d532e141373559b0ad32a0bf4986503678b6986 for an attempt.

Reply:
Aha, nevermind. I forgot that we do have the special handling for dictionaries already.

Reply:
Let's leave the current implementation then, since it is less unsafe than previously believed.

Reply:
It seems it didn't require additional work and even simplified the codegen slightly, but I can then not include the change for object in dictionaries here